gateway/ln-gateway/src/utils.rs

```diff
@@ -5,23 +5,22 @@ use tracing::info;
 
 /// Retry an operation util the operation succeeds, OR
 /// The maximum number of attempts are made without success
-pub async fn retry<F, R, T>(
+pub async fn retry<F, Fut, T>(
     op_name: String,
     op_fn: F,
     wait: Duration,
     max_retries: u32,
 ) -> Result<T, anyhow::Error>
 where
-    F: Fn() -> R,
-    R: Future<Output = Result<T, anyhow::Error>>,
+    F: Fn() -> Fut,
+    Fut: Future<Output = Result<T, anyhow::Error>>,
 {
-    let mut att = 0;
+    let mut retries = 0;
     loop {
         match op_fn().await {
             Ok(result) => return Ok(result),
             Err(e) => {
-                att += 1;
-                if att > max_retries {
+                if retries >= max_retries {
```

Comment:
https://llewellynfalco.blogspot.com/2016/02/dont-use-greater-than-sign-in.html

Reply:
that is a fantastic post. thanks! will adhere to it in the future